### PR TITLE
Add table aws_health_event closes #1161

### LINKED ISSUE #1161: Add table aws_health_event
There are several api paths that are read only but informative and can be used to easily track upcoming events outside of the aws health console

one example would be to query for any upcoming events using https://docs.aws.amazon.com/health/latest/APIReference/API_DescribeEvents.html
and filtering by status: upcoming

**References**

https://docs.aws.amazon.com/health/latest/APIReference/API_Operations.html

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
Add example SQL query results here (please include the input queries as well)
```
</details>

### CODE REVIEW

**Comment by ParthaI** [comment]
Need to test the optional quals queries

**Comment by cbruno10** [comment]
This PR is still under review (but difficult to test since we don't have direct access to the API in our accounts).

File: aws/table_aws_health_event.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+				Description: "The most recent status of the event. Possible values are open, closed, and upcoming.",
+				Type:        proto.ColumnType_STRING,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Arn"),
```
Comment (joelcox22) [inline aws/table_aws_health_event.go:90]:
```suggestion
				Transform:   transform.FromField("Arn").Transform(arnToAkas),
```

**Review by joelcox22 — COMMENTED** [review]
I can assist with testing - this would be a very useful table for me :)

I'm very new to both steampipe and go though, so I won't be much help for a proper code review.

Added a comment suggesting a minor change that seemed to be required to get this mostly working.

side-note: another table based on https://docs.aws.amazon.com/cli/latest/reference/health/describe-affected-entities.html would make this much more useful, as users could write a query to join these health events to an ec2 instance, rds database, etc and get additional context / tags from there

**Comment by joelcox22** [comment]
For some reason, steampipe seems to hang when running a query on this table:

```
> select * from aws_health_event where status_code='open'
⠸ Loading results:  64
```

If I then hit ctrl+c to get it to stop querying, then all the results pop up.

**Comment by joelcox22** [comment]
Here's a log from a simple test query if it helps (with my account id replaced with `<redacted>`)

```json
> select * from aws_<redacted>.aws_health_event limit 10
[
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:us-east-1::event/REDSHIFT/AWS_REDSHIFT_OPERATIONAL_ISSUE/AWS_REDSHIFT_OPERATIONAL_ISSUE_F35B7_8D69B32257C"
  ],
  "arn": "arn:aws:health:us-east-1::event/REDSHIFT/AWS_REDSHIFT_OPERATIONAL_ISSUE/AWS_REDSHIFT_OPERATIONAL_ISSUE_F35B7_8D69B32257C",
  "availability_zone": null,
  "end_time": "2022-09-29T07:42:00+10:00",
  "event_scope_code": "PUBLIC",
  "event_type_category": "issue",
  "event_type_code": "AWS_REDSHIFT_OPERATIONAL_ISSUE",
  "last_updated_time": "2022-09-29T07:42:00+10:00",
  "partition": "aws",
  "region": "global",
  "service": "REDSHIFT",
  "start_time": "2022-09-29T06:39:11+10:00",
  "status_code": "closed"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:global::event/ACM/AWS_ACM_OPERATIONAL_NOTIFICATION/AWS_ACM_OPERATIONAL_NOTIFICATION_dcb8427cd8d81e0b7ef9a55a07edb4cdaf162ac9bf1ea0614f680fa55413310d"
  ],
  "arn": "arn:aws:health:global::event/ACM/AWS_ACM_OPERATIONAL_NOTIFICATION/AWS_ACM_OPERATIONAL_NOTIFICATION_dcb8427cd8d81e0b7ef9a55a07edb4cdaf162ac9bf1ea0614f680fa55413310d",
  "availability_zone": null,
  "end_time": null,
  "event_scope_code": "ACCOUNT_SPECIFIC",
  "event_type_category": "accountNotification",
  "event_type_code": "AWS_ACM_OPERATIONAL_NOTIFICATION",
  "last_updated_time": "2022-09-21T03:47:48+10:00",
  "partition": "aws",
  "region": "global",
  "service": "ACM",
  "start_time": "2022-09-16T13:20:00+10:00",
  "status_code": "open"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:us-west-2::event/SSM/AWS_SSM_OPERATIONAL_ISSUE/AWS_SSM_OPERATIONAL_ISSUE_2E109_FEB7A7EEF12"
  ],
  "arn": "arn:aws:health:us-west-2::event/SSM/AWS_SSM_OPERATIONAL_ISSUE/AWS_SSM_OPERATIONAL_ISSUE_2E109_FEB7A7EEF12",
  "availability_zone": null,
  "end_time": "2022-09-28T10:50:21+10:00",
  "event_scope_code": "PUBLIC",
  "event_type_category": "issue",
  "event_type_code": "AWS_SSM_OPERATIONAL_ISSUE",
  "last_updated_time": "2022-09-28T10:50:21+10:00",
  "partition": "aws",
  "region": "global",
  "service": "SSM",
  "start_time": "2022-09-28T10:11:32+10:00",
  "status_code": "closed"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:me-central-1::event/S3/AWS_S3_OPERATIONAL_ISSUE/AWS_S3_OPERATIONAL_ISSUE_6E6FF_0ECE3D6DE49"
  ],
  "arn": "arn:aws:health:me-central-1::event/S3/AWS_S3_OPERATIONAL_ISSUE/AWS_S3_OPERATIONAL_ISSUE_6E6FF_0ECE3D6DE49",
  "availability_zone": null,
  "end_time": "2022-10-07T09:24:19+11:00",
  "event_scope_code": "PUBLIC",
  "event_type_category": "issue",
  "event_type_code": "AWS_S3_OPERATIONAL_ISSUE",
  "last_updated_time": "2022-10-07T09:24:19+11:00",
  "partition": "aws",
  "region": "global",
  "service": "S3",
  "start_time": "2022-10-07T09:00:48+11:00",
  "status_code": "closed"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:ap-south-1::event/EC2/AWS_EC2_OPERATIONAL_ISSUE/AWS_EC2_OPERATIONAL_ISSUE_4F0AA_7ACE799DFCF"
  ],
  "arn": "arn:aws:health:ap-south-1::event/EC2/AWS_EC2_OPERATIONAL_ISSUE/AWS_EC2_OPERATIONAL_ISSUE_4F0AA_7ACE799DFCF",
  "availability_zone": null,
  "end_time": "2022-09-22T13:35:38+10:00",
  "event_scope_code": "PUBLIC",
  "event_type_category": "issue",
  "event_type_code": "AWS_EC2_OPERATIONAL_ISSUE",
  "last_updated_time": "2022-09-22T13:35:38+10:00",
  "partition": "aws",
  "region": "global",
  "service": "EC2",
  "start_time": "2022-09-22T13:17:19+10:00",
  "status_code": "closed"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:us-west-2::event/APIGATEWAY/AWS_APIGATEWAY_OPERATIONAL_ISSUE/AWS_APIGATEWAY_OPERATIONAL_ISSUE_B105D_513B551D3F0"
  ],
  "arn": "arn:aws:health:us-west-2::event/APIGATEWAY/AWS_APIGATEWAY_OPERATIONAL_ISSUE/AWS_APIGATEWAY_OPERATIONAL_ISSUE_B105D_513B551D3F0",
  "availability_zone": null,
  "end_time": "2022-09-29T08:40:04+10:00",
  "event_scope_code": "PUBLIC",
  "event_type_category": "issue",
  "event_type_code": "AWS_APIGATEWAY_OPERATIONAL_ISSUE",
  "last_updated_time": "2022-09-29T08:40:04+10:00",
  "partition": "aws",
  "region": "global",
  "service": "APIGATEWAY",
  "start_time": "2022-09-29T03:13:00+10:00",
  "status_code": "closed"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:global::event/ROUTE53/AWS_ROUTE53_OPERATIONAL_ISSUE/AWS_ROUTE53_OPERATIONAL_ISSUE_EA360_00146154BA6"
  ],
  "arn": "arn:aws:health:global::event/ROUTE53/AWS_ROUTE53_OPERATIONAL_ISSUE/AWS_ROUTE53_OPERATIONAL_ISSUE_EA360_00146154BA6",
  "availability_zone": null,
  "end_time": "2022-09-19T23:25:28+10:00",
  "event_scope_code": "PUBLIC",
  "event_type_category": "issue",
  "event_type_code": "AWS_ROUTE53_OPERATIONAL_ISSUE",
  "last_updated_time": "2022-09-19T23:25:28+10:00",
  "partition": "aws",
  "region": "global",
  "service": "ROUTE53",
  "start_time": "2022-09-19T20:57:56+10:00",
  "status_code": "closed"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:me-central-1::event/MULTIPLE_SERVICES/AWS_MULTIPLE_SERVICES_OPERATIONAL_ISSUE/AWS_MULTIPLE_SERVICES_OPERATIONAL_ISSUE_FD350_F276A0D2CF1"
  ],
  "arn": "arn:aws:health:me-central-1::event/MULTIPLE_SERVICES/AWS_MULTIPLE_SERVICES_OPERATIONAL_ISSUE/AWS_MULTIPLE_SERVICES_OPERATIONAL_ISSUE_FD350_F276A0D2CF1",
  "availability_zone": null,
  "end_time": "2022-10-07T09:24:07+11:00",
  "event_scope_code": "PUBLIC",
  "event_type_category": "issue",
  "event_type_code": "AWS_MULTIPLE_SERVICES_OPERATIONAL_ISSUE",
  "last_updated_time": "2022-10-07T09:24:06+11:00",
  "partition": "aws",
  "region": "global",
  "service": "MULTIPLE_SERVICES",
  "start_time": "2022-10-07T08:39:30+11:00",
  "status_code": "closed"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:us-west-2::event/GRAFANA/AWS_GRAFANA_OPERATIONAL_ISSUE/AWS_GRAFANA_OPERATIONAL_ISSUE_058E3_C1703F3AEFA"
  ],
  "arn": "arn:aws:health:us-west-2::event/GRAFANA/AWS_GRAFANA_OPERATIONAL_ISSUE/AWS_GRAFANA_OPERATIONAL_ISSUE_058E3_C1703F3AEFA",
  "availability_zone": null,
  "end_time": "2022-10-05T19:30:01+11:00",
  "event_scope_code": "PUBLIC",
  "event_type_category": "issue",
  "event_type_code": "AWS_GRAFANA_OPERATIONAL_ISSUE",
  "last_updated_time": "2022-10-05T19:30:01+11:00",
  "partition": "aws",
  "region": "global",
  "service": "GRAFANA",
  "start_time": "2022-10-05T18:15:46+11:00",
  "status_code": "closed"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:sa-east-1::event/MULTIPLE_SERVICES/AWS_MULTIPLE_SERVICES_OPERATIONAL_ISSUE/AWS_MULTIPLE_SERVICES_OPERATIONAL_ISSUE_2D6EE_2884D869C1E"
  ],
  "arn": "arn:aws:health:sa-east-1::event/MULTIPLE_SERVICES/AWS_MULTIPLE_SERVICES_OPERATIONAL_ISSUE/AWS_MULTIPLE_SERVICES_OPERATIONAL_ISSUE_2D6EE_2884D869C1E",
  "availability_zone": null,
  "end_time": "2022-09-29T23:57:56+10:00",
  "event_scope_code": "PUBLIC",
  "event_type_category": "issue",
  "event_type_code": "AWS_MULTIPLE_SERVICES_OPERATIONAL_ISSUE",
  "last_updated_time": "2022-09-29T23:57:56+10:00",
  "partition": "aws",
  "region": "global",
  "service": "MULTIPLE_SERVICES",
  "start_time": "2022-09-29T23:20:41+10:00",
  "status_code": "closed"
 }
]
```

**Comment by misraved** [comment]
@joelcox22 sorry for the radio silence on this PR. Thanks a lot for testing it out 👍.

@ParthaI has made some updates to the code, by migrating it to the latest v2 version of the SDK. Could you please rebuild this PR and test out the changes?

Once again thanks for testing out this table and providing feedback 👍.

**Comment by joelcox22** [comment]
Hi @misraved, thanks and no problem, thanks for looping back to this 😄 

Tested again with latest changes from this PR - [still same minor change I mentioned before required](https://github.com/turbot/steampipe-plugin-aws/pull/1167#discussion_r990712715) but works with that:

```diff
-Transform:   transform.FromField("Arn"),
+Transform:   transform.FromField("Arn").Transform(arnToAkas),
```

I no longer hit the [issue I mentioned](https://github.com/turbot/steampipe-plugin-aws/pull/1167#issuecomment-1272423856) earlier with steampipe hanging while running the query (even when running this over ~80 AWS accounts with no LIMIT clause) 👍🏻 - just guessing the AWS SDK version upgrade may have helped with that?

```json
> select * from aws_<redacted>.aws_health_event limit 10
[
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:ap-southeast-2::event/RDS/AWS_RDS_OPERATIONAL_NOTIFICATION/AWS_RDS_OPERATIONAL_NOTIFICATION_745ee9771e349e1c2ae1188a625793fe92287a37635b55ecdc2c68c5f3048939"
  ],
  "arn": "arn:aws:health:ap-southeast-2::event/RDS/AWS_RDS_OPERATIONAL_NOTIFICATION/AWS_RDS_OPERATIONAL_NOTIFICATION_745ee9771e349e1c2ae1188a625793fe92287a37635b55ecdc2c68c5f3048939",
  "availability_zone": null,
  "end_time": null,
  "event_scope_code": "ACCOUNT_SPECIFIC",
  "event_type_category": "accountNotification",
  "event_type_code": "AWS_RDS_OPERATIONAL_NOTIFICATION",
  "last_updated_time": "2022-11-11T20:10:24+11:00",
  "partition": "aws",
  "region": "global",
  "service": "RDS",
  "start_time": "2022-11-11T19:05:00+11:00",
  "status_code": "open"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:ap-southeast-2::event/EC2/AWS_EC2_PERSISTENT_INSTANCE_RETIREMENT_SCHEDULED/AWS_EC2_PERSISTENT_INSTANCE_RETIREMENT_SCHEDULED_ff1f5c87-ed5f-4a07-97ed-af04cb0545d0"
  ],
  "arn": "arn:aws:health:ap-southeast-2::event/EC2/AWS_EC2_PERSISTENT_INSTANCE_RETIREMENT_SCHEDULED/AWS_EC2_PERSISTENT_INSTANCE_RETIREMENT_SCHEDULED_ff1f5c87-ed5f-4a07-97ed-af04cb0545d0",
  "availability_zone": null,
  "end_time": "2022-11-28T11:00:00+11:00",
  "event_scope_code": "ACCOUNT_SPECIFIC",
  "event_type_category": "scheduledChange",
  "event_type_code": "AWS_EC2_PERSISTENT_INSTANCE_RETIREMENT_SCHEDULED",
  "last_updated_time": "2022-11-13T10:12:53+11:00",
  "partition": "aws",
  "region": "global",
  "service": "EC2",
  "start_time": "2022-11-28T11:00:00+11:00",
  "status_code": "upcoming"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:ap-southeast-2::event/EC2/AWS_EC2_PERSISTENT_INSTANCE_RETIREMENT_SCHEDULED/AWS_EC2_PERSISTENT_INSTANCE_RETIREMENT_SCHEDULED_0f860dc4-a984-4f4c-aa1d-1c0e27caf4d9"
  ],
  "arn": "arn:aws:health:ap-southeast-2::event/EC2/AWS_EC2_PERSISTENT_INSTANCE_RETIREMENT_SCHEDULED/AWS_EC2_PERSISTENT_INSTANCE_RETIREMENT_SCHEDULED_0f860dc4-a984-4f4c-aa1d-1c0e27caf4d9",
  "availability_zone": null,
  "end_time": "2022-12-05T11:00:00+11:00",
  "event_scope_code": "ACCOUNT_SPECIFIC",
  "event_type_category": "scheduledChange",
  "event_type_code": "AWS_EC2_PERSISTENT_INSTANCE_RETIREMENT_SCHEDULED",
  "last_updated_time": "2022-11-21T08:06:50+11:00",
  "partition": "aws",
  "region": "global",
  "service": "EC2",
  "start_time": "2022-12-05T11:00:00+11:00",
  "status_code": "upcoming"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:ap-southeast-2::event/SDK/AWS_SDK_SECURITY_NOTIFICATION/AWS_SDK_SECURITY_NOTIFICATION_b7b0865830e75c313c47f43f29fd4d3130825a5847863b4f04f738d410fc6675"
  ],
  "arn": "arn:aws:health:ap-southeast-2::event/SDK/AWS_SDK_SECURITY_NOTIFICATION/AWS_SDK_SECURITY_NOTIFICATION_b7b0865830e75c313c47f43f29fd4d3130825a5847863b4f04f738d410fc6675",
  "availability_zone": null,
  "end_time": null,
  "event_scope_code": "ACCOUNT_SPECIFIC",
  "event_type_category": "accountNotification",
  "event_type_code": "AWS_SDK_SECURITY_NOTIFICATION",
  "last_updated_time": "2022-11-19T00:05:37+11:00",
  "partition": "aws",
  "region": "global",
  "service": "SDK",
  "start_time": "2022-11-18T23:50:00+11:00",
  "status_code": "open"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:ap-southeast-2::event/RDS/AWS_RDS_OPERATIONAL_NOTIFICATION/AWS_RDS_OPERATIONAL_NOTIFICATION_a6d91186682ad42993d4d84705fba3c4951d00729fe8b9140ecfae8fcfe79e88"
  ],
  "arn": "arn:aws:health:ap-southeast-2::event/RDS/AWS_RDS_OPERATIONAL_NOTIFICATION/AWS_RDS_OPERATIONAL_NOTIFICATION_a6d91186682ad42993d4d84705fba3c4951d00729fe8b9140ecfae8fcfe79e88",
  "availability_zone": null,
  "end_time": null,
  "event_scope_code": "ACCOUNT_SPECIFIC",
  "event_type_category": "accountNotification",
  "event_type_code": "AWS_RDS_OPERATIONAL_NOTIFICATION",
  "last_updated_time": "2022-11-05T07:51:57+11:00",
  "partition": "aws",
  "region": "global",
  "service": "RDS",
  "start_time": "2022-11-05T07:30:00+11:00",
  "status_code": "open"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:us-east-1::event/TRANSFER/AWS_TRANSFER_OPERATIONAL_ISSUE/AWS_TRANSFER_OPERATIONAL_ISSUE_2425A_18F932ED970"
  ],
  "arn": "arn:aws:health:us-east-1::event/TRANSFER/AWS_TRANSFER_OPERATIONAL_ISSUE/AWS_TRANSFER_OPERATIONAL_ISSUE_2425A_18F932ED970",
  "availability_zone": null,
  "end_time": "2022-11-09T08:01:31+11:00",
  "event_scope_code": "PUBLIC",
  "event_type_category": "issue",
  "event_type_code": "AWS_TRANSFER_OPERATIONAL_ISSUE",
  "last_updated_time": "2022-11-09T08:01:31+11:00",
  "partition": "aws",
  "region": "global",
  "service": "TRANSFER",
  "start_time": "2022-11-09T07:28:00+11:00",
  "status_code": "closed"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:ap-southeast-2::event/RDS/AWS_RDS_AURORA_HARDWARE_MAINTENANCE_SCHEDULED/AWS_RDS_AURORA_HARDWARE_MAINTENANCE_SCHEDULED_APS2_20221127_20221204-<redacted>"
  ],
  "arn": "arn:aws:health:ap-southeast-2::event/RDS/AWS_RDS_AURORA_HARDWARE_MAINTENANCE_SCHEDULED/AWS_RDS_AURORA_HARDWARE_MAINTENANCE_SCHEDULED_APS2_20221127_20221204-<redacted>",
  "availability_zone": null,
  "end_time": "2022-12-04T17:00:00+11:00",
  "event_scope_code": "ACCOUNT_SPECIFIC",
  "event_type_category": "scheduledChange",
  "event_type_code": "AWS_RDS_AURORA_HARDWARE_MAINTENANCE_SCHEDULED",
  "last_updated_time": "2022-11-13T16:13:26+11:00",
  "partition": "aws",
  "region": "global",
  "service": "RDS",
  "start_time": "2022-11-27T16:00:00+11:00",
  "status_code": "upcoming"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:global::event/SECURITY/AWS_SECURITY_TLS_DEPRECATION_NOTIFICATION/AWS_SECURITY_TLS_DEPRECATION_NOTIFICATION_2ff68a373e83ac0bda8e30261f5cc07d6bc4441f626eff06bcfdd5a4743492bc"
  ],
  "arn": "arn:aws:health:global::event/SECURITY/AWS_SECURITY_TLS_DEPRECATION_NOTIFICATION/AWS_SECURITY_TLS_DEPRECATION_NOTIFICATION_2ff68a373e83ac0bda8e30261f5cc07d6bc4441f626eff06bcfdd5a4743492bc",
  "availability_zone": null,
  "end_time": null,
  "event_scope_code": "ACCOUNT_SPECIFIC",
  "event_type_category": "accountNotification",
  "event_type_code": "AWS_SECURITY_TLS_DEPRECATION_NOTIFICATION",
  "last_updated_time": "2022-11-18T12:02:31+11:00",
  "partition": "aws",
  "region": "global",
  "service": "SECURITY",
  "start_time": "2022-11-18T11:10:00+11:00",
  "status_code": "open"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:global::event/CLOUDFORMATION/AWS_CLOUDFORMATION_OPERATIONAL_NOTIFICATION/AWS_CLOUDFORMATION_OPERATIONAL_NOTIFICATION_b544eec553def53fa07e2106054359e9114a8bfc053509a2a481ba71890926be"
  ],
  "arn": "arn:aws:health:global::event/CLOUDFORMATION/AWS_CLOUDFORMATION_OPERATIONAL_NOTIFICATION/AWS_CLOUDFORMATION_OPERATIONAL_NOTIFICATION_b544eec553def53fa07e2106054359e9114a8bfc053509a2a481ba71890926be",
  "availability_zone": null,
  "end_time": null,
  "event_scope_code": "ACCOUNT_SPECIFIC",
  "event_type_category": "accountNotification",
  "event_type_code": "AWS_CLOUDFORMATION_OPERATIONAL_NOTIFICATION",
  "last_updated_time": "2022-11-06T09:24:09+11:00",
  "partition": "aws",
  "region": "global",
  "service": "CLOUDFORMATION",
  "start_time": "2022-11-02T17:40:00+11:00",
  "status_code": "open"
 },
 {
  "_ctx": {
   "connection_name": "aws_<redacted>"
  },
  "account_id": "<redacted>",
  "akas": [
   "arn:aws:health:us-east-1::event/ACM/AWS_ACM_RENEWAL_STATE_CHANGE/AWS_ACM_RENEWAL_STATE_CHANGE-06863f08-adb0-4afd-a9b1-42bd95439aa7"
  ],
  "arn": "arn:aws:health:us-east-1::event/ACM/AWS_ACM_RENEWAL_STATE_CHANGE/AWS_ACM_RENEWAL_STATE_CHANGE-06863f08-adb0-4afd-a9b1-42bd95439aa7",
  "availability_zone": null,
  "end_time": "2022-11-11T13:42:37+11:00",
  "event_scope_code": "ACCOUNT_SPECIFIC",
  "event_type_category": "scheduledChange",
  "event_type_code": "AWS_ACM_RENEWAL_STATE_CHANGE",
  "last_updated_time": "2022-11-11T13:42:38+11:00",
  "partition": "aws",
  "region": "global",
  "service": "ACM",
  "start_time": "2022-11-11T13:42:37+11:00",
  "status_code": "closed"
 }
]
```

Let me know if there's anything specific you want me to try - seems good to me from quick tests so far though.

(also, `availability_zone` field is `null` in all the above examples - I checked specifically with `select * from aws_<redacted>.aws_health_event where availability_zone is not null`, and can confirm it is populating that field appropriately, there's just no relevant examples in the limited dataset I pasted above)

**Comment by misraved** [comment]
Thanks @joelcox22 for such in-depth feedback 😄. 

> side-note: another table based on https://docs.aws.amazon.com/cli/latest/reference/health/describe-affected-entities.html would make this much more useful, as users could write a query to join these health events to an ec2 instance, rds database, etc and get additional context / tags from there

Agreed we would definitely need to add the above API to get valuable information on the resources being affected by such events.

@ParthaI could you please verify if the above API would suit more as a column, or a table? As per my understanding, since the API calls support pagination, I am drawn to believe that the data set is going to be big.

**Review by misraved — APPROVED** [review]

**Pull request merged**